content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js

```diff
@@ -604,49 +612,12 @@
             mutations.forEach(function(mutation) {
                 if (mutation.type === "attributes") {
                     var isAltCheckboxChecked = $(altCheckboxSelector).attr("checked");
-                    var alertIcon = $(altInputAlertIconSelector);
-                    var assetTab = $(assetTabSelector);
-                    var assetTabAlertIcon = $(assetTabAlertIconSelector);
-                    if (mutation.attributeName === "data-seeded-value") {
-                        if (isAltCheckboxChecked) {
-                            if ($(altInputSelector).val()) {
-                                if (alertIcon.length) {
-                                    $(altInputSelector).removeClass("is-invalid");
-                                    alertIcon.hide();
-                                    assetTab.removeClass("is-invalid");
-                                    assetTabAlertIcon.hide();
-                                }
-                            } else {
-                                if (alertIcon.length) {
-                                    $(altInputSelector).addClass("is-invalid");
-                                    alertIcon.show();
-                                    assetTab.addClass("is-invalid");
-                                    assetTabAlertIcon.show();
-                                }
-                            }
-                        }
-                    }
-
-                    if (mutation.attributeName === "disabled") {
-                        if ($(altInputSelector).val()) {
-                            if (alertIcon.length) {
-                                $(altInputSelector).removeClass("is-invalid");
-                                alertIcon.hide();
-                                assetTab.removeClass("is-invalid");
-                                assetTabAlertIcon.hide();
-                            }
-                        }
+                    if (mutation.attributeName === "data-seeded-value" && isAltCheckboxChecked) {
+                        toggleAltTextValidity();
                     }
 
-                    if (mutation.attributeName === "invalid") {
-                        if (!$(altInputSelector).val()) {
-                            if (alertIcon.length) {
-                                $(altInputSelector).addClass("is-invalid");
-                                alertIcon.show();
-                                assetTab.addClass("is-invalid");
-                                assetTabAlertIcon.show();
-                            }
-                        }
+                    if (["invalid", "disabled"].includes(mutation.attributeName)) {
+                        toggleAltTextValidity();
```

Comment:
Is that possible that both 
```
mutation.attributeName === "data-seeded-value" && isAltCheckboxChecked
```
and
```
["invalid", "disabled"].includes(mutation.attributeName)
```
will be `true`?

If that is the case - do we want to execute `toggleAltTextValidity()` twice?
Maybe simply do:
```
if ((mutation.attributeName === "data-seeded-value" && isAltCheckboxChecked) ||
    ["invalid", "disabled"].includes(mutation.attributeName)
) {
    toggleAltTextValidity();
}
```

Reply:
They can't be simultaneously true, because mutation.attributeName is a string.
And it wouldn't hurt if it triggered twice anyways.

Reply:
I still think
```
if ((mutation.attributeName === "data-seeded-value" && isAltCheckboxChecked) ||
    ["invalid", "disabled"].includes(mutation.attributeName)) {
    toggleAltTextValidity();
}
```
would be a better approach.

If `mutation.attributeName === "data-seeded-value" && isAltCheckboxChecked` then we would execute `toggleAltTextValidity()` and return from the function.
But with your approach after executing `toggleAltTextValidity()` instead of immediately returning you will always evaluate `["invalid", "disabled"].includes(mutation.attributeName)` for no reason since we know that `mutation.attributeName === "data-seeded-value"`

What's worse - this statement will potentially be evaluated multiple times since this code is contained within `mutations.forEach` loop.

Reply:
makes sense. will do.